pytest_splunk_addon/standard_lib/fields_tests/test_templates.py

```diff
@@ -405,21 +408,34 @@ def test_datamodels(
         extracted_tags = fields_from_splunk.get("tag", "")
         extracted_tags = extracted_tags.strip("][").split(", ")
         extracted_tags = [tag.replace("'", "") for tag in extracted_tags]
-
-        extracted_datamodels = []
-        for datamodel, tags in dict_datamodel_tag.items():
-            if all(tag in extracted_tags for tag in tags):
-                extracted_datamodels.append(datamodel)
-
         self.logger.debug(f"Tags extracted from Splunk {extracted_tags}")
 
+        matched_datamodels = {
+            dm: tags
+            for dm, tags in dict_datamodel_tag.items()
+            if all(tag in extracted_tags for tag in tags)
+        }
+        assigned_datamodels = {
+            dm: tags
+            for dm, tags in matched_datamodels.items()
+            if not any(
+                set(tags).issubset(set(matched_tags)) and dm != matched_datamodel
+                for matched_datamodel, matched_tags in matched_datamodels.items()
+            )
+        }
+
         record_property("search", search)
 
-        missing_datamodels = [dm for dm in datamodels if dm not in extracted_datamodels]
-        wrong_datamodels = [dm for dm in extracted_datamodels if dm not in datamodels]
+        missing_datamodels = {
+            dm: tags for dm, tags in datamodels.items() if dm not in assigned_datamodels
+        }
```

Comment:
datamodels is  a list. Maybe?:

```suggestion
            missing_datamodels = [dm for dm in datamodels if dm not in assigned_datamodels]
```

Reply:
If applies `wrong_datamodels` might be a list too